pkg/bytecode/compiler.go

```diff
@@ -0,0 +1,120 @@
+package bytecode
+
+import (
+	"errors"
+	"fmt"
+
+	"evylang.dev/evy/pkg/parser"
+)
+
+// ErrUndefinedVar is returned when a variable name cannot
+// be resolved in the symbol table.
+var ErrUndefinedVar = errors.New("undefined variable")
+
+// Compiler is responsible for turning a parsed evy program into
+// bytecode.
+type Compiler struct {
+	constants    []value
+	instructions Instructions
+	symbolTable  *SymbolTable
+}
+
+// Bytecode represents raw evy bytecode.
+type Bytecode struct {
+	Constants    []value
+	Instructions Instructions
```

Comment:
This Bytecode type is represented as the separate fields in two other places - `Compiler` and `VM`. It is not all that clear to me why you don't just use the `Bytecode` struct in those two places and just copy the single value rather than the individual fields.

Reply:
I guess not using it avoids a layer on indirection in the compiler and vm, e.g. you have to go and look at bytecode to understand that it has instructions/constants and it's only really a wrapper for passing a bag of data from compiler -> vm. I don't feel particularly strongly about it so can change if you think that it is better.

Reply:
keep it as is. let's see where it goes.